### PR TITLE
Fix next block addr

### PR DESCRIPTION
**Pull request by echoedit**
Signed-off-by: Mengchi Zhang zhan2308@purdue.edu

### CODE REVIEW

File: src/cuda-sim/ptx_ir.cc [hunk 1]
```diff
@@ -458,7 +458,7 @@ void function_info::connect_basic_blocks( ) //iterate across m_basic_blocks of f
          if( pI->has_pred() ) {
             printf("GPGPU-Sim PTX: Warning detected predicated return/exit.\n");
             // if predicated, add link to next block
-            unsigned next_addr = pI->get_m_instr_mem_index() + 1;
+            unsigned next_addr = pI->get_m_instr_mem_index() + pI->inst_size();
```
Comment (tgrogers) [inline src/cuda-sim/ptx_ir.cc]: Remind me - how did you observe this issue?
What effect was this having? If what you have done is right then the previous implementation was actually pretty broken right? Anything with a predicated exit is basically broken?

Comment (tgrogers) [inline src/cuda-sim/ptx_ir.cc]: Also - I am not sure why your regressions are failing - can you update your branch to use the latest gpgpu-sim?

Comment (echoedit) [inline src/cuda-sim/ptx_ir.cc]: Ok. Let me update and apply this patch again

Comment (tgrogers) [inline src/cuda-sim/ptx_ir.cc]: Also - please update the description of what you have done. Clearly what you have done is simple, but we need some context here. Ideally we would have an automated test case that excites the issue in the old code and is obviously fixed by the new code.
In lieu of that, please just write a paragraph describing how you observed the problem, in enough detail that someone could produce a test-case if they wanted to.

Comment (echoedit) [inline src/cuda-sim/ptx_ir.cc]: I have created a new pull request for newest dev branch, so that I will close this pull request.

**Pull request closed without merging**